apps/E2E/src/common/BasePage.ts

```diff
@@ -209,33 +213,82 @@ export abstract class BasePage {
     });
   }
 
-  /* Scrolls to the specified or primary UI test element until it is displayed. */
-  async scrollToTestElement(component?: WebdriverIO.Element): Promise<void> {
-    const ComponentToScrollTo = component ?? (await this._primaryComponent);
-    if (await ComponentToScrollTo.isDisplayed()) {
+  /* Scrolls to the specified or primary UI test element until it is displayed.
+   * Either the component identifier or the component itself can be passed in as a parameter.
+   * On Android only the component identifier can be passed in.
+   * If no parameter is passed in, the primary component is used.
+   */
+  async scrollToTestElement(component?: WebdriverIO.Element | string): Promise<void> {
+    let componentIdentifier: string;
+    let componentToScrollTo: WebdriverIO.Element;
+
+    if (typeof component == 'string') {
+      componentIdentifier = component ?? this._primaryComponentName;
```

Comment:
I don't think we need the `?? this._primaryComponentName`? If the parameter is a string, then we know the user passed in a string value (non-null value). We should have a separate check to see if the parameter is null or not, and if it is, then set equal to `this._primaryComponentName`

Reply:
`
if(typeof component === 'undefined')
{
  componentToScrollTo = await this._primaryComponent;
}
else if(if (typeof component == 'string') {}
else { componentToScrollTo = component};
}

Reply:
Updated